jerry-main/main-unix.c

```diff
@@ -513,25 +573,28 @@ main (int argc,
 
   ret_value = jerry_create_undefined ();
 
-  for (int i = 0; i < exec_snapshots_count; i++)
+  if (jerry_is_feature_enabled (JERRY_FEATURE_SNAPSHOT_EXEC))
   {
-    size_t snapshot_size;
-    const uint8_t *snapshot_p = read_file (exec_snapshot_file_names[i], &snapshot_size);
-
-    if (snapshot_p == NULL)
+    for (int i = 0; i < exec_snapshots_count; i++)
     {
-      ret_value = jerry_create_error (JERRY_ERROR_COMMON, (jerry_char_t *) "");
-    }
-    else
-    {
-      ret_value = jerry_exec_snapshot ((void *) snapshot_p,
-                                       snapshot_size,
-                                       true);
-    }
+      size_t snapshot_size;
+      const uint8_t *snapshot_p = read_file (exec_snapshot_file_names[i], &snapshot_size);
 
-    if (jerry_value_has_error_flag (ret_value))
-    {
-      break;
+      if (snapshot_p == NULL)
+      {
+        ret_value = jerry_create_error (JERRY_ERROR_COMMON, (jerry_char_t *) "");
```

Comment:
The value of the `snapshot_p` assigned by the the `read_file` function before this condition. As I saw it, this function is add an error message before returns with a NULL and because in this case we breaks from the for loop here, we reach a condition which checks if the 'ret_value' value is an error value. In case of an error it calls the 'print_unhandled_exception' function which also gave a detailed error message.